x/ccv/provider/keeper/power_shaping.go

```diff
@@ -679,30 +641,69 @@ func (k Keeper) UpdatePrioritylist(ctx sdk.Context, consumerId string, priorityl
 	}
 }
 
+// // FilterAndSortPriorityList filters the priority list to include only validators that can validate the chain
+// // by ensuring they are present in the validators list.
+// // It then sorts the filtered list of validators in descending order based on their power.
+// func (k Keeper) FilterAndSortPriorityList(priorityList []string, validators []types.ConsensusValidator) []types.ConsensusValidator {
+// 	validatorMap := make(map[string]types.ConsensusValidator)
+// 	for _, v := range validators {
+// 		validatorMap[string(v.ProviderConsAddr)] = v
+// 	}
+
+// 	filteredPriority := make([]types.ConsensusValidator, 0)
+// 	addedAddresses := make(map[string]bool)
+
+// 	for _, address := range priorityList {
+// 		if validator, exists := validatorMap[address]; exists {
+// 			if !addedAddresses[address] {
+// 				filteredPriority = append(filteredPriority, validator)
+// 				addedAddresses[address] = true
+// 			}
+// 		}
+// 	}
+
+// 	sort.Slice(filteredPriority, func(i, j int) bool {
+// 		return filteredPriority[i].Power > filteredPriority[j].Power
+// 	})
+
+// 	return filteredPriority
+// }
```

Comment:
don't forget to remove this commented out code if not needed

Reply:
It's removed now.